addons/event/controllers/main.py

```diff
@@ -23,3 +24,55 @@ def event_ics_file(self, event, **kwargs):
             ('Content-Length', len(content)),
             ('Content-Disposition', content_disposition('%s.ics' % event.name))
         ])
+
+    @route(['''/event/tickets_dl'''], type='http', auth="public")
+    def event_tickets_dl(self, dl_hash, event_id, registration_ids, badge_mode=False):
+        """ Returns a pdf response, containing all tickets for attendees in registration_ids for event_id.
+        They can be filtered out based on conditions in `_fetch_registrations_for_event_tickets_dl`
+        Redirects to the portal if not registration is valid / hash is invalid / parameters are missing.
+        This route is used in links in emails to attendees, as well as in registration confirmation screens.
+
+        :param badge_mode: boolean, True to use template of foldable badge instead of full page ticket.
+        :param dl_hash: string hash used to access the tickets.
+        :param event_id: the id of the prompted event. Only its attendees will be considered.
+        :param registration_ids: ids of event.registrations of which tickets are generated
+        """
+        event_id = int(event_id) if event_id else False
+        registration_ids = ast.literal_eval(registration_ids or '[]')
```

Comment:
Good to know! I've seen you used unquote_plus in appointment. Can you walk me through the reason you use it there? (I do not think this is relevant here) Thx!

Reply:
It's because the JS part now use `encodeURIComponent` for security reason as we call `document.location` and could lead to code injection.
For example the date_time param the controller receives now look like this: `2023-06-23 14%3A00%3A00` after the encode, so we need to retransform the value we get with unquote_plus in the controller

Reply:
Just seeing this one now ^^ We don't pass in JS so it's ok. What I meant was someone could do for example: tweak the url to have `javascript:alert('evil laugh')` which will then execute in JS without the encodeURIComponent. The unquote_plus is just to get the value decoded if it was encoded in JS

Reply:
Ok, I think I see, thanks for the info